niederbayern_oberpfalz.py

```diff
@@ -111,88 +110,113 @@ def parse_url(url, today=False):
         'ZTVG':  'vegan',
         'ZTW':   'Wild'
     }
-    #canteen.setLegendData(legend)
 
+    # Create regular expressions for categories
     hg = re.compile("^HG[1-9]$")
     b = re.compile("^B[1-9]$")
     n = re.compile("^N[1-9]$")
 
-    #for w in 0, 1:
-    for w in [0]:
-        kw = (date.today() + timedelta(weeks=w)).isocalendar()[1]
-        try:
-            f = urlopen('%(location)s/%(isoweek)d.csv' %
-                        {'location': url, 'isoweek': kw})
-        except HTTPError as e:
-            if e.code == 404:
-                continue
+    # Get current isoweek and try to get the data
+    # On error 404 return empty feed
+    # At most locations the data doesn't exist on term break
+    kw = date.today().isocalendar()[1]
+    try:
+        f = urlopen('%(location)s/%(isoweek)d.csv' %
+                    {'location': url, 'isoweek': kw})
+    except HTTPError as e:
+        if e.code == 404:
+            return canteen.toXMLFeed()
+        else:
+            raise e
+
+    # Decode data from ISO charset
+    f = f.read().decode('iso8859-1')
+
+    # Set roles for prices
+    roles = ('student', 'employee', 'other')
+
+    # Read csv data and skip the csv header
+    mealreader = reader(f.splitlines(), delimiter=';')
+    next(mealreader)
+    for row in mealreader:
+        mdate = row[0]
+        category = row[2]
+        mname = row[3]
+        mtype = row[4]
+        prices = [row[6], row[7], row[8]]
+
+        # determine category for the current meal
+        if category == 'Suppe':
+            pass
+        elif hg.match(category):
+            category = 'Hauptgerichte'
+        elif b.match(category):
+            category = 'Beilagen'
+        elif n.match(category):
+            category = 'Nachspeisen'
+        else:
+            raise RuntimeError('Unknown category: ' + str(category))
+
+        # Extract the notes from brackets in the meal name
+        # Remove the brackets, notes and improve readability
+        notes = []
+        bpos = mname.find(')')
+        while bpos != -1:
+            apos = mname.find('(')
+            # Extract notes from current brackets and avoid empty notes
+            for i in mname[apos+1:bpos].split(','):
+                if i:
+                    notes.append(i)
+            # Check if brackets are at the end of the meal name
+            if bpos == len(mname)-1:
+                # Remove brackets and break bracket loop
+                mname = mname[:apos]
+                bpos = -1
             else:
-                raise e
-        f = f.read().decode('iso8859-1')
+                # Remove current brackets, improve readability
+                # and find the next brackets
+                mname = mname[:apos] + ' und ' + mname[bpos+1:]
```

Comment:
NIT: This generally results in the name containing too many spaces, eg `Jägerschnitzel (3,A,G,I)Bio Schwäbische Spätzle (A,C)` from [UNI-R/37.csv](http://www.stwno.de/infomax/daten-extern/csv/UNI-R/37.csv) results in two spaces between `Jägerschnitzel` and `und Bio Schwäbische Spätzle`.
I would suggest something like `mname[:apos].rstrip() + ' und ' + mname[bpos+1:].lstrip()`.